anta/tests/routing/bgp.py

```diff
@@ -1404,3 +1444,145 @@ def test(self) -> None:
+
+            @model_validator(mode="after")
+            def validate_inputs(self: BaseModel) -> BaseModel:
+                """Validate the inputs provided to the BgpAfi class.
+
+                If afi is either ipv4 or ipv6, safi must be provided.
+
+                If afi is not ipv4 or ipv6, safi must not be provided and vrf must be default.
+                """
+                if self.afi in ["ipv4", "ipv6"]:
+                    if self.safi is None:
+                        msg = "'safi' must be provided when afi is ipv4 or ipv6"
+                        raise ValueError(msg)
+                elif self.safi is not None:
+                    msg = "'safi' must not be provided when afi is not ipv4 or ipv6"
+                    raise ValueError(msg)
+                elif self.vrf != "default":
+                    msg = "'vrf' must be default when afi is not ipv4 or ipv6"
+                    raise ValueError(msg)
+                return self
+
+    def render(self, template: AntaTemplate) -> list[AntaCommand]:
+        """Render the template for each BGP address family in the input list."""
+        commands = []
+        for afi in self.inputs.address_families:
+            if template == VerifyBGPPeerPrefixes.commands[0] and afi.afi in ["ipv4", "ipv6"] and afi.safi != "sr-te":
+                commands.append(template.render(afi=afi.afi, safi=afi.safi, vrf=afi.vrf))
+
+            # For SR-TE SAFI, the EOS command supports sr-te first then ipv4/ipv6
+            elif template == VerifyBGPPeerPrefixes.commands[0] and afi.afi in ["ipv4", "ipv6"] and afi.safi == "sr-te":
+                commands.append(template.render(afi=afi.safi, safi=afi.afi, vrf=afi.vrf))
+            elif template == VerifyBGPPeerPrefixes.commands[1] and afi.afi not in ["ipv4", "ipv6"]:
+                commands.append(template.render(afi=afi.afi))
+        return commands
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyBGPPeerPrefixes."""
+        failures: dict[Any, Any] = {}
+
+        for command, input_entry in zip(self.instance_commands, self.inputs.address_families):
+            command_output = command.json_output
+            afi = input_entry.afi
+            safi = input_entry.safi
+            afi_vrf = input_entry.vrf
+            peers = input_entry.peers
+
+            # Update failures dictionary for `afi`, later on removing the same if no failure found.
+            if not failures.get(afi):
+                failures[afi] = {}
+
+            # Verify peer details.
+            if not (peer_details := get_value(command_output, f"vrfs..{afi_vrf}..peers", separator="..")):
+                failures[afi].update({safi: "Peers not configured"})
+                if not safi:
+                    failures[afi] = "Peers not configured"
+                continue
+
+            # Verify the received and accepted prefix(s).
+            failure_logs = _get_inconsistent_peers(peer_details, peers)
```

Comment:
please separate out the variable name by input or actual peers

Reply:
Updated. Thanks!!